File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    
    answer = 1 #자기 자신
    
    for i in range(1, n//2 + 1):
        total = 0
        
        while total < n:
            total += i
            
            if total == n:
                answer += 1
                break
            
            i += 1
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    for (int i=1; i<=n; i++){
        int number = i;
        int sum = 0;
        
        while (sum <= n) {            
            sum += number;
            number += 1;
            
            if (sum == n) {
                answer += 1;
                break;
            }
        }
    }
    
    return answer;
}